jha/src/views/enrolment/ContactInfoPage.vue

```diff
@@ -424,6 +435,21 @@ export default {
       this.isMailSame = false;
     },
   },
+  watch: {
+    mailCountry: function(value) {
+      // don't clear province if the change is from loading from store
+      if (this.mailCountryOld !== "_") {
+        if(this.mailCountryOld === "Canada" && value !== "Canada") {
+          // don't keep a Canadian province if not in canada
+          this.mailProvince = "";
+        } else if (this.mailCountryOld !== "Canada" && value === "Canada") {
+          // if changing to canada, a canadian province must be selected by default.
+          this.mailProvince = "British Columbia";
+        }
+      }
+      this.mailCountryOld = value;
```

Comment:
You can get the old value via a second parameter to the watch function. https://vuejs.org/v2/api/#watch

Reply:
Updated with second parameter